BUG when compiling code in mac OS

Note: The issue is introduced by the following commit
```
commit 198c7fa3e34c71bc075bcfb067da80242e74bddf
Author: Vladimir Kobal <vlad@prokk.net>
Date:   Mon May 13 15:45:34 2019 +0300

    Add AWS Kinesis backend (#5914)
```

Following the instructions here for MacOS: https://docs.netdata.cloud/packaging/installer/#macos

When i try to build from source, compilation breaks:
```
$ autoreconf -ivf && ./configure  --prefix="${NETDATA_PREFIX}/usr" --sysconfdir="${NETDATA_PREFIX}/etc" --localstatedir="${NETDATA_PREFIX}/var" --with-zlib --with-math --with-user="${NETDATA_USER}"
.
.
.
.
make
.
.
.
  CC       registry/registry_url.o
  CC       database/rrdcalc.o
  CC       database/rrdcalctemplate.o
  CC       database/rrddim.o
  CC       database/rrddimvar.o
  CC       database/rrdfamily.o
  CC       database/rrdhost.o
  CC       database/rrd.o
  CC       database/rrdset.o
  CC       database/rrdsetvar.o
  CC       database/rrdvar.o
  CC       streaming/rrdpush.o
  CC       collectors/statsd.plugin/statsd.o
  CC       web/server/web_client.o
database/rrdset.c:1265:168: warning: format specifies type 'long' but the argument has type '__darwin_suseconds_t' (aka 'int') [-Wformat]
  ...too old data (last updated at %ld.%ld, last collected at %ld.%ld). Resetting it. Will not store the next entry.", st->name, st->last_updated.tv_sec, st->last_updated.tv_usec, st->last_coll...
                                       ~~~                                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~
                                       %d
database/../daemon/../libnetdata/log/log.h:85:71: note: expanded from macro 'info'
#define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
                                                                      ^~~~
database/rrdset.c:1265:226: warning: format specifies type 'long' but the argument has type '__darwin_suseconds_t' (aka 'int') [-Wformat]
  ...at %ld.%ld). Resetting it. Will not store the next entry.", st->name, st->last_updated.tv_sec, st->last_updated.tv_usec, st->last_collected_time.tv_sec, st->last_collected_time.tv_usec);
            ~~~                                                                                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            %d
database/../daemon/../libnetdata/log/log.h:85:71: note: expanded from macro 'info'
#define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
                                                                      ^~~~
  CC       web/server/web_server.o
  CC       web/server/web_client_cache.o
  CC       web/server/static/static-threaded.o
  CC       collectors/macos.plugin/plugin_macos.o
2 warnings generated.
  CC       collectors/macos.plugin/macos_sysctl.o
  CC       collectors/macos.plugin/macos_mach_smi.o
  CC       collectors/macos.plugin/macos_fw.o
  CCLD     cups.plugin
collectors/macos.plugin/macos_fw.c:118:35: warning: address of array 'diskstat.name' will always evaluate to 'true' [-Wpointer-bool-conversion]
            if(unlikely(!diskstat.name || !*diskstat.name)) {
                        ~~~~~~~~~~^~~~
./config.h:345:41: note: expanded from macro 'unlikely'
#define unlikely(x) __builtin_expect(!!(x), 0)
                                        ^
1 warning generated.
  GEN      netdata
undef: ___CFConstantStringClassReference
undef: _kCFAllocatorDefault
undef: _IOIteratorReset
undef: _CFNumberGetValue
undef: _IOServiceMatching
undef: _IOServiceGetMatchingServices
undef: _IORegistryEntryGetChildEntry
undef: _IOIteratorNext
undef: _CFStringGetCString
undef: _IORegistryEntryCreateCFProperties
undef: _CFDictionaryGetValue
undef: _IOMasterPort
undef: _CFRelease
undef: _IOObjectRelease
Undefined symbols for architecture x86_64:
  "___CFConstantStringClassReference", referenced from:
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      CFString in lto.o
      ...
  "_kCFAllocatorDefault", referenced from:
      _macos_main in lto.o
  "_IOIteratorReset", referenced from:
      _macos_main in lto.o
  "_CFNumberGetValue", referenced from:
      _macos_main in lto.o
  "_IOServiceMatching", referenced from:
      _macos_main in lto.o
  "_IOServiceGetMatchingServices", referenced from:
      _macos_main in lto.o
  "_IORegistryEntryGetChildEntry", referenced from:
      _macos_main in lto.o
  "_IOIteratorNext", referenced from:
      _macos_main in lto.o
  "_CFStringGetCString", referenced from:
      _macos_main in lto.o
  "_IORegistryEntryCreateCFProperties", referenced from:
      _macos_main in lto.o
  "_CFDictionaryGetValue", referenced from:
      _macos_main in lto.o
  "_IOMasterPort", referenced from:
      _macos_main in lto.o
  "_CFRelease", referenced from:
      _macos_main in lto.o
  "_IOObjectRelease", referenced from:
      _macos_main in lto.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [netdata] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
 FAILED   
```

_Originally posted by @paulkatsoulakis in https://github.com/netdata/netdata/issues/5737#issuecomment-493469798_

--- a/Makefile.am
+++ b/Makefile.am
@@ -508,9 +508,9 @@ netdata_LDADD = \
 	$(NETDATA_COMMON_LIBS) \
 	$(NULL)
 if ENABLE_BACKEND_KINESIS
-    netdata_LINK = $(CXXLD) $(CXXFLAGS) -o $@
+    netdata_LINK = $(CXXLD) $(CXXFLAGS) $(LDFLAGS) -o $@
 else
-    netdata_LINK = $(CCLD) $(CFLAGS) -o $@
+    netdata_LINK = $(CCLD) $(CFLAGS) $(LDFLAGS) -o $@
 endif
 
 if ENABLE_PLUGIN_APPS

